src/shared/components/MapTo.tsx

```diff
@@ -226,7 +226,14 @@ export const MapToDropdownBasic = ({
   children,
   config,
 }: MapToDropdownBasic) => (
-  <Dropdown visibleElement={<Button variant="tertiary">{children}</Button>}>
+  <Dropdown
+    // eslint-disable-next-line react/no-unstable-nested-components
```

Comment:
I see, this is conceptually the same though. I'm wondering if we should just disable this rule globally if we have to work around it like this?

Reply:
I only see lint exception in one other place and looking at the [eslint page](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md) it says rather bluntly:

> When Not To Use It
If you are not interested in preventing bugs related to re-creation of the nested components or do not care about optimization of virtual DOM.

I'm thinking either refactoring or case by case exception in case it introduces some bug that becomes difficult to understand.

Reply:
At the same time, the page also says "By default component creation is allowed inside component props only if prop name starts with `render`", which makes me think that it's not necessarily a bad thing, or at least they recognise it is unavoidable some times